analyses/fusion_filtering/02-fusion-filtering.R

```diff
@@ -105,15 +117,26 @@ fusion_filtering_QC<-function(standardFusioncalls=standardFusioncalls,readingFra
   # @param spanningFragCountFilter An integer threshold for (SpanningFragCount - JunctionReadCount)
   # @param expressionFilter An integer threshold for TPM/FPKM filter
   # @return Standardized fusion calls filtered to pass QC and remove calls with insufficient read-support and annotation red-flags
-  
-  if( reathroughFilter & any(grepl('read.*through|NEIGHBORS',standardFusioncalls$annots,ignore.case = TRUE))){
-    # Gather read throughs from standardized fusion calls  
+
+  if( readthroughFilter & any(grepl('read.*through|NEIGHBORS',standardFusioncalls$annots,ignore.case = TRUE))){
+    # Gather read throughs from standardized fusion calls
     rts <- standardFusioncalls[grep('read.*through|NEIGHBORS',standardFusioncalls$annots,ignore.case = TRUE),"FusionName"]
-    # Reverse of read throughs to capture 
+    # Reverse of read throughs to capture
     rts.rev <- unique(unlist(lapply(strsplit(rts, '--'), FUN = function(x) paste0(x[2],'--',x[1]))))
-    # Combine read through and reverse fusion genes 
+    # Combine read through and reverse fusion genes
     rts <- unique(c(rts, rts.rev))
-    standardFusioncalls <- standardFusioncalls[-which(standardFusioncalls$FusionName %in% rts),]
+    # remove read throughs even if distance is not same in intergenic fusions
+    rts<-unlist(lapply(rts,function(x) rm_between(x, "(", ")", extract = F)))
+    rts<-data.frame("readThroughs"=rts)
+    standardFusioncalls<-standardFusioncalls[-which(unlist(lapply(standardFusioncalls$FusionName,function(x) rm_between(x, "(", ")", extract = F))) %in% rts$readThroughs),]
+
+    # TO-DO
+    # Get readthroughs from other callers
+    if (!file.exists("scratch/readthroughs.tsv")){
```

Comment:
If you wanted to save this into a different directory, like the `results` folder in this module, you would have to change this in 3 places. Can you instead accept the read-through file as a command line argument? You can set it to `scratch/readthroughs.tsv` by default if you would like.

Reply:
I removed saving this file since now I merge the standard fusion calls and remove read throughs from combined set so this step to save read throughs in a file is not required.
However there is a TO-DO I need to add a logic in the readthrough filter to only remove annotated read throughs if the fusion is not seen in cancer specific fusion databases. I'll add the logic after I commit all your suggestions just so you can review that separately.. trying to not make a mess of commits like last time :D